core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -120,6 +120,35 @@ private static Block fixedWidthSequence(long start, long stop, long step, FixedW
         return blockBuilder.build();
     }
 
+    private static int getLength(long start, long stop, long step)
+    {
+        // handle the case when start and stop are either both positive, or both negative
+        if ((start > 0 && stop > 0) || (start < 0 && stop < 0)) {
+            return checkMaxEntry((stop - start) / step + 1);
+        }
```

Comment:
```suggestion
        // handle the case when start and stop are either both positive, or both negative
        if ((start > 0 && stop > 0) || (start < 0 && stop < 0)) {
            long steps = (stop - start) / step;
            checkMaxEntry(steps);
            return checkMaxEntry(steps + 1);
        }
```

Reply:
I did similar to the suggestion, is it ok for you